lib/internal/module.js

```diff
@@ -3,11 +3,34 @@
 exports = module.exports = {
   makeRequireFunction,
   stripBOM,
-  addBuiltinLibsToObject
+  addBuiltinLibsToObject,
+  applyTransforms
 };
 
 exports.requireDepth = 0;
 
+const transforms = [];
+
+function addTransform(transform) {
+  transforms.push(transform)
+}
+function removeTransform(transform) {
+  const index = transforms.indexOf(transform)
+  if (index >= 0) {
+    transforms.splice(index, 1)
+  }
+}
+
+function applyTransforms(content, filename) {
+  return transforms.reduce((content, transform) => {
+    const next = transform(content, filename);
+    if (typeof next !== 'string') {
+      throw new Error('Module transforms must return the modified content')
+    }
+    return next;
+  }, content);
+}
+
```

Comment:
Certainly code duplication is a thing to avoid generally, but the code to wrap an event emitter around it has just as much surface area as the custom implementation itself, without even taking into account all the extra stuff running inside the event emitter implementation. It's also a lot more difficult for someone unfamiliar with the code to understand at first glance. Were this something more complicated, an event emitter implementation might make more sense, but it has basically zero advantage here.

Also, the event emitter implementation makes the require call 2.5-3 times slower and has 11 times the memory footprint. It's not premature optimization, it's carefully considered and diligently measured optimization. 😉